server/service/integration_ds_only_test.go

```diff
@@ -0,0 +1,63 @@
+package service
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type integrationDSTestSuite struct {
+	withDS
+	suite.Suite
+}
+
+func TestIntegrationDSTestSuite(t *testing.T) {
+	testingSuite := new(integrationDSTestSuite)
+	testingSuite.s = &testingSuite.Suite
+	suite.Run(t, testingSuite)
+}
+
+func (s *integrationDSTestSuite) SetupSuite() {
+	s.withDS.SetupSuite("integrationDSTestSuite")
+}
+
+func (s *integrationDSTestSuite) TestLicenseExpiration() {
+	testCases := []struct {
+		name             string
+		tier             string
+		expiration       time.Time
+		shouldHaveHeader bool
+	}{
+		{"basic expired", fleet.TierBasic, time.Now().Add(-24 * time.Hour), true},
+		{"basic not expired", fleet.TierBasic, time.Now().Add(24 * time.Hour), false},
+		{"core expired", fleet.TierCore, time.Now().Add(-24 * time.Hour), false},
+		{"core not expired", fleet.TierCore, time.Now().Add(24 * time.Hour), false},
+	}
+
+	createTestUsers(s.T(), s.ds)
+	for _, tt := range testCases {
+		s.Run(tt.name, func() {
+			t := s.T()
+
+			license := &fleet.LicenseInfo{Tier: tt.tier, Expiration: tt.expiration}
+			_, server := RunServerForTestsWithDS(t, s.ds, TestServerOpts{License: license, SkipCreateTestUsers: true})
```

Comment:
The deferred `server.Close()` could be registered in `t.Cleanup` so the caller doesn't have to remember to defer it.